File: suites/05_ml_primitives/comparisons/softmax.cpp
#include <cstdio>
static long long softmax_like(long long width, long long rounds) {
    long long total = 0;
    for (long long r = 0; r < rounds; ++r) {
        long long partial = 0;
        for (long long l = 0; l < width; ++l)
            partial += ((l * 11) + r) % 251;
        total += partial;
    }
    return total;
}
int main() { printf("%lld\n", softmax_like(4096, 4000)); return 0; }

File: suites/05_ml_primitives/comparisons/softmax.py
def softmax_like(width: int, rounds: int) -> int:
    total = 0
    for r in range(rounds):
        partial = 0
        for l in range(width):
            partial += ((l * 11) + r) % 251
        total += partial
    return total
if __name__ == "__main__":
    print(softmax_like(4096, 4000))
